internal/handler/wallet/import.go

```diff
@@ -2,98 +2,52 @@ package wallet
 
 import (
 	"github.com/go-openapi/runtime/middleware"
-	"lukechampine.com/blake3"
-
 	"github.com/massalabs/thyra-plugin-massa-wallet/api/server/models"
 	"github.com/massalabs/thyra-plugin-massa-wallet/api/server/restapi/operations"
-
-	"github.com/btcsuite/btcutil/base58"
+	"github.com/massalabs/thyra-plugin-massa-wallet/pkg/password"
+	"github.com/massalabs/thyra-plugin-massa-wallet/pkg/privateKey"
 	"github.com/massalabs/thyra-plugin-massa-wallet/pkg/wallet"
 )
 
-// checkDuplicate checks that the same wallet doesn't already exist.
-func checkDuplicate(nickname string) middleware.Responder {
-	_, err := wallet.Load(nickname)
-	if err == nil {
-		return operations.NewRestWalletImportInternalServerError().WithPayload(
-			&models.Error{
-				Code:    errorAlreadyExists,
-				Message: "Error: a wallet with the same nickname already exists.",
-			})
-	}
+func NewImport(pkPrompt privateKey.Asker, pwdPrompt password.Asker) operations.RestWalletImportHandler {
+	return &wImport{pwdPrompt: pwdPrompt, pkPrompt: pkPrompt}
+}
 
-	return nil
+type wImport struct {
+	pwdPrompt password.Asker
+	pkPrompt  privateKey.Asker
 }
 
-// decodeWalletAttributes decodes all wallet attributes.
-func decodeWalletAttributes(rawPrivateKey string, rawPublicKey string, rawSalt string, rawNonce string) (privKey []byte, pubKey []byte, salt [16]byte, nonce [12]byte, resp middleware.Responder) {
-	resp = operations.NewRestWalletCreateUnprocessableEntity()
+func (c *wImport) Handle(params operations.RestWalletImportParams) middleware.Responder {
+	walletName := params.Nickname
 
-	privKey, _, err := base58.CheckDecode(rawPrivateKey)
+	password, err := c.pwdPrompt.Ask(walletName)
 	if err != nil {
-		return
+		return ImportWalletError(errorImportWalletCanceled, errorImportWalletCanceled)
 	}
 
-	pubKey, _, err = base58.CheckDecode(rawPublicKey)
+	privateKey, err := c.pkPrompt.Ask()
 	if err != nil {
-		return
+		return ImportWalletError(errorImportWalletCanceled, errorImportWalletCanceled)
 	}
 
-	unsizedSalt, _, err := base58.CheckDecode(rawSalt)
-	if err != nil {
-		return
+	_, err = wallet.Load(walletName)
+	if err == nil {
+		return ImportWalletError(errorImportNickNameAlreadyTaken, errorImportNickNameAlreadyTaken)
 	}
 
-	copy(salt[:], unsizedSalt)
-
-	unsizedNonce, _, err := base58.CheckDecode(rawNonce)
+	newWallet, err := wallet.Imported(walletName, privateKey, password)
 	if err != nil {
-		return
+		return ImportWalletError(err.Error(), err.Error())
 	}
 
-	copy(nonce[:], unsizedNonce)
-
-	resp = nil
-	return
+	return New(newWallet)
 }
 
-// failure generates an internal server error based on given error.
-func failure(err error) middleware.Responder {
-	return operations.NewRestWalletCreateInternalServerError().WithPayload(
+func ImportWalletError(code string, message string) middleware.Responder {
+	return operations.NewRestWalletImportInternalServerError().WithPayload(
```

Comment:
The only real internl server error is line 41. All other are due to bad information sent by the caller or duplicate record. You should then use the proper 4XX return code.